src/lexers/javascript-lexer.js

```diff
@@ -1,39 +1,57 @@
-import * as acorn from 'acorn-jsx'
-import injectAcornObjectRestSpread from "acorn-object-rest-spread/inject"
+import * as acorn from 'acorn'
+import injectAcornStage3 from "acorn-stage3/inject"
 import injectAcornEs7 from "acorn-es7"
 import * as walk from 'acorn/dist/walk'
 import BaseLexer from './base-lexer'
 
+const WalkerBase = Object.assign({}, walk.base, {
+  Import(node, st, c) {
+    // We need this catch, but we don't need the catch to do anything.
+  }
+})
+
 export default class JavascriptLexer extends BaseLexer {
   constructor(options = {}) {
     super(options)
 
-    this.acornOptions = { sourceType: 'module', ...options.acorn }
+    this.acornOptions = { 
+      sourceType: 'module', 
+      ecmaVersion: 9,
+      ...options.acorn,
+      plugins: {
+        stage3: true,
+        es7: true,
+        ...(options.acorn ? options.acorn.plugins : {})
+      }
+    }
+
     this.functions = options.functions || ['t']
     this.attr = options.attr || 'i18nKey'
-  }
-
-  extract(content) {
-    const that = this
 
-    let localAcorn = acorn
+    this.acorn = acorn
+    this.WalkerBase = WalkerBase
 
     if (this.acornOptions.plugins) {
-      if (this.acornOptions.plugins.es7) {
-        injectAcornEs7(localAcorn)
+      if (this.acornOptions.plugins.stage3) {
+        this.acorn = injectAcornStage3(this.acorn)
       }
-      if (this.acornOptions.plugins.objectRestSpread) {
-        injectAcornObjectRestSpread(localAcorn)
+      if (this.acornOptions.plugins.es7) {
+        injectAcornEs7(this.acorn)
       }
     }
+  }
+
+  extract(content) {
+    const that = this
 
     walk.simple(
-      localAcorn.parse(content, this.acornOptions),
+      this.acorn.parse(content, this.acornOptions),
       {
         CallExpression(node) {
           that.expressionExtractor.call(that, node)
         }
-      }
+      },
+      this.WalkerBase
```

Comment:
These changes are made here so that we can build a walker for JavaScript, then inherit them better into the JSX lexer (instead of duplicating config).

this.acorn is the acorn that's modified for JavaScript (acorn injections)
this.WalkerBase is the acorn that's modified for JavaScript (walk extensions)